module/zfs/zil.c

```diff
@@ -1870,6 +1871,25 @@ zil_lwb_write_close(zilog_t *zilog, lwb_t *lwb)
 
 	dmu_tx_commit(tx);
 
+	/*
+	 * Acquire config lock for the lwb to issue it later.  If we have a
+	 * queue of already closed parent lwbs already holding config lock,
+	 * we can not block here now, or it will be a deadlock.
+	 */
+	if (ilwbs && !list_is_empty(ilwbs)) {
+		if (!spa_config_tryenter(spa, SCL_STATE, lwb, RW_READER)) {
+			lwb_t *tlwb;
+			while ((tlwb = list_remove_head(ilwbs)) != NULL)
+				zil_lwb_write_issue(zilog, tlwb);
+			spa_config_enter(spa, SCL_STATE, lwb, RW_READER);
```

Comment:
we don't have to call `spa_config_enter` for each LWB that we issue?

each LWB will call `spa_config_exit` via `zil_lwb_flush_vdevs_done`, right? so, I think as-is, we might have more `exit` calls than `enter` calls?

Reply:
If spa_config_tryenter() succeed, sure we don't need spa_config_enter() for that lwb any more.  Otherwise the count must match.  For previous lwbs it is already called.

Reply:
> For previous lwbs it is already called.

oh, right..